R/fonts.R

```diff
@@ -381,12 +381,24 @@ hash <- function(x) {
   digest::digest(x, algo = "spookyhash")
 }
 
+
+find_cache_dir <- function(pkg) {
+  # In R 4.0 and above, CRAN wants us to use the new tools::R_user_dir().
+  # If not present, fall back to rappdirs::user_cache_dir().
+  R_user_dir <- .getNamespace('tools')$R_user_dir
```

Comment:
Why not `asNamespace()` -- is this faster?

Reply:
`asNamespace(x)` wraps around `getNamespace(x)`, which does a little more than `.getNamespace(x)`. Maybe use `getNamespace("tools")` to be safer?

Reply:
For future reference: `getNamespace()` is better than `.getNamespace()`, since the former will cause `tools` to be loaded if it wasn't already. If `Rscript` is used (instead of `R`), `tools` won't be loaded by default:


```
$ Rscript -e "loadedNamespaces()"
[1] "compiler"  "graphics"  "utils"     "grDevices" "stats"     "datasets" 
[7] "methods"   "base"     
```


However, if `getNamespace('tools')` is called, then it will cause the package to load:
```
$ Rscript -e "getNamespace('tools'); loadedNamespaces()"
<environment: namespace:tools>
[1] "compiler"  "graphics"  "tools"     "utils"     "grDevices" "stats"    
[7] "datasets"  "methods"   "base"     
```